services/offline/src/lib/online-status.js

```diff
@@ -0,0 +1,40 @@
+import throttle from 'lodash/throttle'
+import { useState, useEffect, useCallback } from 'react'
+
+/**
+ * Returns the browser's online status. Updates in response to 'online' and
+ * 'offline' events. By default, throttles updates to once every second to
+ * avoid UI flicker, but that delay can be configured with the
+ * `options.throttleDelay` param.
+ *
+ * TODO: Add option to periodically ping server to check online status.
+ *
+ * @param {Object} [options]
+ * @param {Number} [options.throttleDelay] - Timeout delay to throttle updates, in ms
+ * @returns {Object} `{ online, offline }` booleans. Each is the opposite of the other.
+ */
+export function useOnlineStatus(options) {
+    const [online, setOnline] = useState(navigator.onLine)
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const updateState = useCallback(
+        throttle(
```

Comment:
I think you want `lodash.debounce` instead of `lodash.throttle` here since we want the UI to always show the latest value of `navigator.onLine` (but ignore rapid changes in between).

Reply:
I'm okay with that; what do you think about including an update on the leading-edge of the debounce though? I think it will be nicer to skip a delay if the status only changes once 🙂 And what do you think about the 1s delay?

Reply:
Yep leading-edge makes sense and 1s delay seems fine to me.

Reply:
Ultimately switched to trailing-edge debounce after talking with Austin 👍